src/core/lib/event_engine/windows/windows_engine.h

```diff
@@ -104,16 +99,52 @@ class WindowsEventEngine : public EventEngine,
   bool Cancel(TaskHandle handle) override;
 
  private:
-  struct Closure;
+  // State of an active connection.
+  // Managed by a shared_ptr, owned exclusively by the timeout callback and the
+  // OnConnectCompleted callback herein.
+  struct ConnectionState {
+    // everything is guarded by mu;
+    grpc_core::Mutex mu;
+    EventEngine::ConnectionHandle connection_handle ABSL_GUARDED_BY(mu);
+    EventEngine::TaskHandle timer_handle ABSL_GUARDED_BY(mu);
+    EventEngine::OnConnectCallback on_connected_user_callback
+        ABSL_GUARDED_BY(mu);
+    EventEngine::Closure* on_connected ABSL_GUARDED_BY(mu);
+    std::unique_ptr<WinSocket> socket ABSL_GUARDED_BY(mu);
+    EventEngine::ResolvedAddress address ABSL_GUARDED_BY(mu);
+    MemoryAllocator allocator ABSL_GUARDED_BY(mu);
+  };
+
+  // A poll worker which schedules itself unless kicked
+  class IOCPWorker : public EventEngine::Closure {
```

Comment:
Maybe this could be renamed to IOCPWorkClosure ?

Reply:
Done